stl/src/filesystem.cpp

```diff
@@ -850,25 +820,48 @@ struct alignas(long long) _Aligned_file_attrs {
 
     constexpr auto _Get_file_attributes_data =
         __std_fs_stats_flags::_Attributes | __std_fs_stats_flags::_File_size | __std_fs_stats_flags::_Last_write_time;
-    if (_Bitmask_includes(
-            _Flags, _Get_file_attributes_data)) { // caller wants something GetFileAttributesExW might provide
+    if (_Bitmask_includes(_Flags,
+            _Get_file_attributes_data)) { // caller wants something GetFileAttributesExW/FindFirstFileW might provide
         if (_Symlink_attribute_hint == __std_fs_file_attr::_Invalid
             || !_Bitmask_includes(_Symlink_attribute_hint, __std_fs_file_attr::_Reparse_point)
-            || !_Follow_symlinks) { // we might not be a symlink or not following symlinks, so GetFileAttributesExW
-                                    // would return the right answer
-            _Aligned_file_attrs _Aligned_attrs;
-            auto& _Data = _Aligned_attrs._Data;
+            || !_Follow_symlinks) { // we might not be a symlink or not following symlinks, so
+                                    // GetFileAttributesExW/FindFirstFileW would return the right answer
+
+            WIN32_FILE_ATTRIBUTE_DATA _Data;
             if (!GetFileAttributesExW(_Path, GetFileExInfoStandard, &_Data)) {
-                return __std_win_error{GetLastError()};
+                __std_win_error _Last_error{GetLastError()};

```

Comment:
This could be `const`. Its scope could also be limited with braces. (Having `_Last_error` remain in scope isn't very dangerous, but we've seen issues where such variables were incorrectly returned later instead of calling `GetLastError()` anew; limiting scope prevents such issues.)

We could also use `if`-with-initializer - there are a few occurrences of that C++17 feature in the codebase. It would wrap, but it would avoid the extra braces.